Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -426,8 +434,30 @@ void FieldProbe::ComputeDiags (int step)
         for (MyParIter pti(m_probe, lev); pti.isValid(); ++pti)
         {
             const auto getPosition = GetParticlePosition(pti);
-            auto const np = pti.numParticles();
+                auto setPosition = SetParticlePosition(pti);
 
+            auto const np = pti.numParticles();
+            if (do_moving_window_FP && step > warpx.start_moving_window_step && step <= warpx.end_moving_window_step)
+            {
+                int step_diff = step - m_last_compute_step;
+                for (auto ip=0; ip < np; ip++)
+                {
+                    amrex::ParticleReal xp, yp, zp;
+                    getPosition(ip, xp, yp, zp);
+                    if (warpx.moving_window_dir == 0)
+                    {
+                        setPosition(ip, xp+(temp_dist_mult*step_diff), yp, zp);
```

Comment:
lets move the declaration of `temp_dist_mult` before the loops and also multiply in `step_diff`. That way, we save computation in the inner "hot" loop.
```suggestion
                        setPosition(ip, xp+move_dist, yp, zp);
```